collector/collector.go

```diff
@@ -494,7 +494,16 @@ func NewCollector(system, endpoint, prefix string, servers []*CollectedServer) p
 		return newReplicatorCollector(getSystem(system, prefix), servers)
 	}
 	if isJszEndpoint(system) {
-		return newJszCollector(getSystem(system, prefix), endpoint, servers)
+		return newJszCollector(getSystem(system, prefix), endpoint, servers, []string{}, []string{})
 	}
 	return newNatsCollector(getSystem(system, prefix), endpoint, servers)
 }
+
+// NewJszCollector creates a new NATS JetStream Collector.
+func NewJszCollector(
+	endpoint, prefix string,
+	servers []*CollectedServer,
+	streamMetaKeys, consumerMetaKeys []string,
+) prometheus.Collector {
+	return newJszCollector(getSystem(JetStreamSystem, prefix), endpoint, servers, streamMetaKeys, consumerMetaKeys)
+}
```

Comment:
There is a separate code path now for JetStream collectors to be able to supply the meta keys. This seemed better than to pass the arguments around for the other collectors where they wouldn't be used.

This makes JS collectors somewhat of a special case in the code, not sure if you like it that way.